### ISSUE TITLE
When error in script and "Open dominant script on Scene change" is checked, related scene shoud be opened

### ISSUE THREAD

**Issue by kubecz3k** [godotengine/godot#5646]
**Operating system or device - Godot version:**

**Issue description** (what happened, and what was expected):
Was avoiding to open this issue for a very long time, mostly because I don't feel like a lot of people would benefit from this, but it's coming back to me constantly, so here we go:
In general I'm not using script panel to much, it's quite confusing for me. As far as I know for people like me there is an option `Open dominant script on Scene change`, when this option is checked for 90% of time the script you are seeing is the script that is related to opened scene and this is cool.
The problem occurs in those remaining 10% of time. The most common case when this happens is simply at the time when there is an error in the script:
![scene_script_relation](https://cloud.githubusercontent.com/assets/6129594/16733524/2e919490-4782-11e6-9c50-b4ee36c89c3b.png)
In this example the error is inside the script that is related to SplashNew.tscn scene, but what I'm seeing when I'm looking at tabs is "BaseBall.tscn" and this is confusing me regularly.  

**Steps to reproduce:**

**Link to minimal example project** (optional but very welcome):

**Comment by kubecz3k** [comment]
Another issue about "Open dominant script on scene change option": https://github.com/godotengine/godot/issues/3638

**Comment by aaronfranke** [comment]
This probably isn't relevant anymore. The script editor now has a list of open scripts on the left.

![d](https://user-images.githubusercontent.com/1646875/47276579-1cc7f300-d586-11e8-9ed1-29fa260dd4d8.png)

**Comment by kubecz3k** [comment]
If we still have the option then the issue is still relevant

**Comment by MaxStgs** [comment]
Is still actual? I wants to take it. I am thinking that there is problem, when one script can be attached to few scenes, in these way looks like should be opened there is error occurs.

**Comment by KoBeWi** [comment]
You can obtain scene information from the node that the script is attached to. Probably something like `owner.filename` (accessed from the node affected by error) should be the scene path.

**Comment by AkhRani** [comment]
Can you confirm this commit fixes the issue?  I can rebase the commit if that helps.

**Comment by MaxStgs** [comment]
Yes, it fixes for 4.x. But Idk how it works for current version.

**Comment by KoBeWi** [comment]
The setting is called "open dominant _script_" while the issue is asking about the opposite, i.e. opening scene from script. I don't think it makes much sense, at least not tied to this setting.
